webtau-data/src/main/java/com/twosigma/webtau/data/csv/CsvParser.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.data.csv;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.math.BigDecimal;
+import java.util.*;
+
+public class CsvParser {
+    private CsvParser() {
+    }
+
+    public static List<Map<String, Object>> parse(String content) {
+        return parse(Collections.emptyList(), content);
+    }
+
+    public static List<Map<String, Object>> parse(List<String> header, String content) {
+        List<Map<String, Object>> tableData = new ArrayList<>();
+
+        CSVParser csvRecords = readCsvRecords(header, content);
+
+        Collection<String> headerToUse = header.isEmpty() ?
+                csvRecords.getHeaderMap().keySet() :
+                header;
+
+        for (CSVRecord record : csvRecords) {
```

Comment:
Why did they thing that having a class called CSVParser implement iterable?  The assignment a few lines above reads so weird because of it :(